### PR TITLE
Implement NLStoCons

### PR DESCRIPTION
**Pull request by abelsiqueira**
Need to rebase after #115

### CODE REVIEW

**Comment by abelsiqueira** [comment]
This is ready for review

**Comment by abelsiqueira** [comment]
This implements `NLStoCons` which converts an `NLSModel` by setting `F(x) = r` and using the sum of r squares as residual. The model is itself an `NLSModel`.

**Comment by dpo** [comment]
This looks great, but I wonder if we can find a better name than NLStoCons?! I've heard people referring to this transformation as "dualizing" the least-squares problem.

**Comment by abelsiqueira** [comment]
I don't like the name also (though I started with `NLStoNLPwCons`), but I didn't understand the "dualizing".

**Comment by dpo** [comment]
I think "dualizing" simply refers to moving a term to the constraints. But it's not a great name either. How about `FeasibilityNLS`? Or `FeasibilityFormNLS`?

**Comment by abelsiqueira** [comment]
I like `FeasibilityFormNLS`.

File: src/NLPModels.jl [hunk 1]
```diff
@@ -365,6 +365,7 @@ include("autodiff_model.jl")
 include("simple_model.jl")
 include("slack_model.jl")
 include("qn_model.jl")
+include("nls_to_cons.jl")
```
Comment (dpo) [inline src/NLPModels.jl]: Would you mind renaming this file? Thanks !

File: test/runtests.jl [hunk 2]
```diff
@@ -78,3 +78,4 @@ test_view_subarrays()
 include("test_nlsmodels.jl")
 include("nls_consistency.jl")
 include("test_generic_coord.jl")
+include("test_nls_to_cons.jl")
```
Comment (dpo) [inline test/runtests.jl]: And this one

**Comment by dpo** [comment]
Great, thank you! Go ahead and merge after the file renames.

**Pull request merged**